main/acle.md

```diff
@@ -9546,25 +9680,2141 @@ possible to write these operations using normal C arithmetic. For example:
+  
+  
+  ```
+
+#### SQRSHRN, UQRSHRN
+
+Multi-vector saturating rounding shift right narrow and interleave
+
+``` c
+  // Variants are also available for _u16_x2
+  __attribute__((arm_streaming_compatible))
+  svint16_t svqrshrn[_s16_x2](svint32x2_t zn, uint64_t imm);
+  
+  
+  // Variants are also available for _u8_x4
+  __attribute__((arm_streaming))
+  svint8_t svqrshrn[_s8_x4](svint32x4_t zn, uint64_t imm);
+
+
+  // Variants are also available for _u16_x4
+  __attribute__((arm_streaming))
+  svint16_t svqrshrn[_s16_x4](svint64x4_t zn, uint64_t imm);
+  
+  ```
+
+#### SQRSHRU
+
+Multi-vector saturating rounding shift right unsigned narrow
+
+``` c
+  
+  __attribute__((arm_streaming))
+  svuint16_t svsqrshru[_u16_x2](svint32x2_t zn, uint64_t imm);
+  
+  
+  __attribute__((arm_streaming))
+  svuint8_t svsqrshru[_u8_x4](svint32x4_t zn, uint64_t imm);
+
+
+  __attribute__((arm_streaming))
+  svuint16_t svsqrshru[_u16_x4](svint64x4_t zn, uint64_t imm);
+  
+  ```
+
+#### SQRSHRUN
+
+Multi-vector saturating rounding shift right unsigned narrow and interleave
+
+``` c
+  
+  __attribute__((arm_streaming_compatible))
+  svuint16_t svsqrshrun[_u16_x2](svint32x2_t zn, uint64_t imm);
+  
+  
+  // Variants are also available for _u16
+  __attribute__((arm_streaming))
+  svuint8_t svsqrshrun[_u8_x4](svint32x4_t zn, uint64_t imm);
+  
+  ```
+
+#### SQDMULH (single)
+
+Multi-vector signed saturating doubling multiply high
+
+``` c
+  // Variants are also available for _single_s16_x2, _single_s32_x2
+  // and _single_s64_x2
+  __attribute__((arm_streaming))
+  svint8x2_t svsqdmulh[_single_s8_x2](svint8x2_t zdn, svint8_t zm);
+  
+  
+  // Variants are also available for _single_s16_x4, _single_s32_x4
+  // and _single_s64_x4
+  __attribute__((arm_streaming))
+  svint8x4_t svsqdmulh[_single_s8_x4](svint8x4_t zdn, svint8_t zm);
+  
+  ```
+
+#### SQDMULH (multi)
+
+Multi-vector signed saturating doubling multiply high
+
+``` c
+  // Variants are also available for _s16_x2, _s32_x2 and _s64_x2
+  __attribute__((arm_streaming))
+  svint8x2_t svsqdmulh[_s8_x2](svint8x2_t zdn, svint8x2_t zm);
+  
+  
+  // Variants are also available for _s16_x4, _s32_x4 and _s64_x4
+  __attribute__((arm_streaming))
+  svint8x4_t svsqdmulh[_s8_x4](svint8x4_t zdn, svint8x4_t zm);
+  
+  ```
+
+#### WHILEGE, WHILEGT, WHILEHI, WHILEHS, WHILELE, WHILELO, WHILELS, WHILELT
+
+While (resulting in predicate-as-counter). ``vl`` is expected to be 2 or 4.
+
+``` c
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilege_c8(int64_t rn, int64_t rm, uint64_t vl);
+  
+  
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilegt_c8(int64_t rn, int64_t rm, uint64_t vl);
+  
+  
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilehi_c8(uint64_t rn, uint64_t rm, uint64_t vl);
```

Comment:
For SVE, we instead used le/lt/ge/gt for unsigned inputs too, with a suffix to indicate the signedness and size of the values being compared.  This follows the `arm_neon.h` convention of using `vcle` for `<=` between floats, signed integers, and unsigned integers.